consensus/hotstuff/signature/block_signer_decoder_test.go

```diff
@@ -65,30 +65,29 @@ func (s *blockSignerDecoderSuite) Test_RootBlock() {
 	require.Empty(s.T(), ids)
 }
 
-// Test_UnknownBlock tests handling of an unknwon block.
-// At the moment, hotstuff.Committee returns an storage.ErrNotFound for an unknown block,
-// which we expect the `BlockSignerDecoder` to wrap into a `state.UnknownBlockError`
-func (s *blockSignerDecoderSuite) Test_UnknownBlock() {
+// Test_UnexpectedCommitteeException verifies that `BlockSignerDecoder`
+// does _not_ erroneously interpret an unexpected exception from the committee as
+// a sign of an unknown block, i.e. the decoder should _not_ return an `state.UnknownBlockError`
+func (s *blockSignerDecoderSuite) Test_UnexpectedCommitteeException() {
+	exception := errors.New("unexpected exception")
 	*s.committee = *hotstuff.NewDynamicCommittee(s.T())
-	s.committee.On("IdentitiesByBlock", mock.Anything).Return(nil, storage.ErrNotFound)
+	s.committee.On("IdentitiesByEpoch", mock.Anything).Return(nil, exception)
 
 	ids, err := s.decoder.DecodeSignerIDs(s.block.Header)
 	require.Empty(s.T(), ids)
-	require.True(s.T(), state.IsUnknownBlockError(err))
+	require.False(s.T(), state.IsUnknownBlockError(err))
+	require.True(s.T(), errors.Is(err, exception))
 }
 
-// Test_UnexpectedCommitteeException verifies that `BlockSignerDecoder`
-// does _not_ erroneously interpret an unexpecgted exception from the committee as
-// a sign of an unknown block, i.e. the decouder should _not_ return an `state.UnknownBlockError`
-func (s *blockSignerDecoderSuite) Test_UnexpectedCommitteeException() {
-	exception := errors.New("unexpected exception")
+// Test_UnknownEpoch tests handling of a block from an unknown epoch.
+// It should propagate the sentinel error model.ErrViewForUnknownEpoch from Committee.
+func (s *blockSignerDecoderSuite) Test_UnknownEpoch() {
 	*s.committee = *hotstuff.NewDynamicCommittee(s.T())
-	s.committee.On("IdentitiesByBlock", mock.Anything).Return(nil, exception)
+	s.committee.On("IdentitiesByEpoch", mock.Anything).Return(nil, model.ErrViewForUnknownEpoch)
 
 	ids, err := s.decoder.DecodeSignerIDs(s.block.Header)
 	require.Empty(s.T(), ids)
-	require.False(s.T(), state.IsUnknownBlockError(err))
-	require.True(s.T(), errors.Is(err, exception))
+	require.ErrorIs(s.T(), err, model.ErrViewForUnknownEpoch)
 }
 
 // Test_InvalidIndices verifies that `BlockSignerDecoder` returns
```

Comment:
could we correct the typo in the [line 94](https://github.com/onflow/flow-go/blob/45d458fa451c0290512990e33404f0b32fe50e53/consensus/hotstuff/signature/block_signer_decoder_test.go#L94) to
> // signature.InvalidSignerIndicesError **if** the signer indices in the provided header